pkg/asset/cluster/tfvars.go

```diff
@@ -940,7 +940,20 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 			}
 		}
 
-		transitGatewayEnabled := powervsconfig.TransitGatewayEnabledZone(installConfig.Config.Platform.PowerVS.Zone)
+		attachedTG := ""
+		tgConnectionVPCID := ""
+		if installConfig.Config.PowerVS.ServiceInstanceGUID != "" {
+			attachedTG, err = client.GetAttachedTransitGateway(ctx, installConfig.Config.PowerVS.ServiceInstanceGUID)
+			if err != nil {
+				return err
+			}
+			if vpcSubnet != "" {
+				tgConnectionVPCID, err = client.GetTGConnectionVPC(ctx, attachedTG, *vpc.ID)
```

Comment:
Isn't it possible that `vpc` is `nil` so this could panic? The only time when you set the value of `vpc` is if `vpcName != nil`.

Reply:
good catch!! @miyamotoh this should actually be `vpcName` instead of `vpcSubnet`

Reply:
or could also check for `vpc != nil`